### ISSUE TITLE
allow to link a carddav addressbook to Mumble

### ISSUE THREAD

**Issue by Mannshoch** [mumble-voip/mumble#4175]
**Context**
 - Have nice and correct name in Mumble
 - Have possibility to insert contact data into the address book
 - Have possibility to sync this data to other PC

**Describe the feature you have in mind**
Right-click user -> Add to address book 

VCF File content:
IMPP;TYPE=home;PREF=1:mumble:USERNAME@SERVER:PORT

**Describe alternatives you've considered**
Allow right click an User and just rename it.

**Comment by Krzmbrzl** [comment]
So what exactly do you want? A way to set the name of other clients for displaying on your machine?

If so, I think this is a duplicate of #3967

**Comment by Mannshoch** [comment]
I try to propose several Ideas that may work hand in hand. Yes, it seems one of these is #3967

**Comment by Krzmbrzl** [comment]
Could you elaborate on the other aspects. To me it is not really clear yet, why I could need an address book in Mumble :thinking:

**Comment by bendem** [comment]
Correct me if I'm wrong. You want the possibility to save information in a card dav server about someone you met on a mumble server (which server, what their user-id was) ?

So you have an entry in your card dav server with me:

```
name=bendem
phone=number
im=mumble:$host:$port:$user-id(based on certificate?)
```

in your contact software you want to be able to connect to the mumble server specified (mumble:// urls should work for that, ask your contact software maintainers).

And in mumble you would like to see the details from carddav if any.

---

If I understood correctly, the contact software part should be easy enough to implement (online availability is probably hard though).
As for the mumble part, if someone feels like implementing it, it would probably be well received by the team, but I doubt such a niche usage will have a high priority for the core team.

**Comment by Krzmbrzl** [comment]
> As for the mumble part, if someone feels like implementing it, it would probably be well received by the team, but I doubt such a niche usage will have a high priority for the core team.

Agreed.

**Comment by Mannshoch** [comment]
> Could you elaborate on the other aspects. To me it is not really clear yet, why I could need an address book in Mumble thinking

- [ ] Carddav access (like Thunderbird CardBook)
- [ ] Sync List of known User betwween clients and show them
- [ ] Show Name according to the Carddav entry.
- [ ] Show on Mumble online User
- [ ] Show other Messaging possibility like E-Mail 
  - [ ] Allow to send an Invite Mail with a calendar Item

**Comment by streaps** [comment]
this would be used by very few user. I think it's feature creep and to much effort. features not only have to be implemented, but also maintained in the future.

**Comment by Mannshoch** [comment]
@all thanks for you respond. In my opinion it's an important Idea but I'm also not the heavy user you may have in focus with this tool. 
May it's more an idea for an add-on so you could test how people use it?

**Comment by toby63** [comment]
Hmm...
maybe a friends list in the mumble client would be a good thing, also for other usecases, like:
#2560 #1813 #3967 

And then regarding the more specific ideas, it could be enough to be able to access the friendslist via API and/or export the data.
This way the data could be imported to another client or modified by an external script to implement it into adressbooks etc.

**Comment by Krzmbrzl** [comment]
I agree that this sounds very much like the task for a plugin. After #3743 has been merged the creation of general purpose plugins will be possible.

I also think that the scope of this issue goes beyond what Mumble tries to be. Therefore I'll close this issue.

**Comment by Mannshoch** [comment]
If you have somewhere a list of Blueprints for plugins I would like it if you move this Ticket on this list.

**Comment by Krzmbrzl** [comment]
I'm pretty sure we don't have that (yet) as right now plugins aren't even a part of the master branch. This will change soon though :)
